### PR TITLE
Create 'make upload' that can upload to S3

### PR DESCRIPTION
**Pull request by djmitche**
I'm not totally sure about the `.taskcluster.yml` here - I'll need to experiment with some "test" releases.  But the `make upload` works great!

### CODE REVIEW

**Comment by djmitche** [comment]
This is for #49.

**Comment by djmitche** [comment]
```
go build -ldflags " -X github.com/taskcluster/taskcluster-cli/cmds/version.VersionNumber=v0.5.5-104-g20c1e20" -o taskcluster .
upload_release/upload.go:10:2: cannot find package "github.com/aws/aws-sdk-go/aws" in any of:
	/go/src/github.com/taskcluster/taskcluster-cli/vendor/github.com/aws/aws-sdk-go/aws (vendor tree)
	/usr/local/go/src/github.com/aws/aws-sdk-go/aws (from $GOROOT)
	/go/src/github.com/aws/aws-sdk-go/aws (from $GOPATH)
```
so basically `go build .` is building `./...`.

**Comment by t0xicCode** [comment]
Would it make more sense to have the "uploading" aspect be either a shell script or a separate project (I've seen some that can build then upload automatically)?

Having both the source code for the program and the utilities/infra/deployment like it is make me a bit weary, and we end up tracking our deployment dependencies at the same place as our program dependencies (a developer shouldn't need to get the AWS library to develop).

**Comment by djmitche** [comment]
Yeah, I'm specifically avoiding putting the AWS SDK in the vendor.json.  It's not practical with a shell script since it involves interacting with the TC APIs to get the AWS credentials, then interacting with AWS.  I could do it in JS or Python I suppose :)

**Comment by djmitche** [comment]
All that's left is a piece of lint that somehow got merged.  Hopefully with that we should have a green PR.  And AWS SDK isn't pulled down except in this task.

**Comment by djmitche** [comment]
Woo, green!  So, it's not unsuccessful :)

File: _upload_release/upload.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+	creds := credentials.NewStaticCredentials(id, secret, token)
+	sess := session.Must(session.NewSession(aws.NewConfig().
+		WithRegion("us-west-1").
+		WithCredentials(creds)))
+
+	return s3.New(sess), nil
+}
+
+func exitErrorf(msg string, args ...interface{}) {
```
Comment (t0xicCode) [inline _upload_release/upload.go]: This can be replaced by [`log.Fatalf`](https://golang.org/pkg/log/#Fatalf) :)

Comment (djmitche) [inline _upload_release/upload.go]: oh, good point -- I'll make a followup PR (I had copy/pasted that from the AWS SDK examples)

**Pull request merged**